homeassistant/components/cover/brunt.py

```diff
@@ -0,0 +1,182 @@
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/cover.brunt
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME)
+from homeassistant.components.cover import (
+    ATTR_POSITION, CoverDevice,
+    PLATFORM_SCHEMA, SUPPORT_CLOSE,
+    SUPPORT_OPEN, SUPPORT_SET_POSITION
+)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['brunt==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Based on an unofficial Brunt SDK.'
+
+COVER_FEATURES = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
+DEVICE_CLASS = 'window'
+
+STATE_PARTIALLY_OPEN = 'partially open'
+ATTR_REQUEST_POSITION = 'request_position'
+DEFAULT_NAME = 'brunt blind engine'
+NOTIFICATION_ID = 'brunt_notification'
+NOTIFICATION_TITLE = 'Brunt Cover Setup'
+
+CLOSED_POSITION = 0
+OPEN_POSITION = 100
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the brunt platform."""
+    # pylint: disable=no-name-in-module
+    from brunt import BruntAPI
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    bapi = BruntAPI(username=username, password=password)
+    try:
+        things = bapi.getThings()['things']
+        if not things:
+            _LOGGER.error("No things present in account.")
+        else:
+            add_devices([BruntDevice(
+                bapi, thing['NAME'],
+                thing['thingUri']) for thing in things], True)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        _LOGGER.error("%s", ex)
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+
+
+class BruntDevice(CoverDevice):
+    """
+    Representation of a Brunt cover device.
+
+    Contains the common logic for all Brunt devices.
+    """
+
+    def __init__(self, bapi, name, thing_uri):
+        """Init the Brunt device."""
+        self._bapi = bapi
+        self._name = name
+        self._thing_uri = thing_uri
+
+        self._state = None
+        self._available = None
+
+    @property
+    def name(self):
+        """Return the name of the device as reported by tellcore."""
+        return self._name
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._available
+
+    @property
+    def current_cover_position(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return int(self._state.get('currentPosition'))
```

Comment:
This will error if `self._state` is still `None`. Init `self._state` as an empty dict and add a safe default as second argument to `dict.get` here, that will pass the int copy.